hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -397,6 +397,25 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 				ObservedGeneration: nodePool.Generation,
 			})
 		}
+
+		if len(nodePool.Spec.Platform.AWS.SecurityGroups) == 0 &&
```

Comment:
With this logic wouldn't we be always setting the condition to false with "Message: "Waiting for AWS default security group to be created for hosted cluster" for any platform different than aws?

Reply:
nvm the diff got me confused, I see now this is within `if nodePool.Spec.Platform.Type == hyperv1.AWSPlatform {` block